### ISSUE TITLE
Cloudflare tunnel equivalent of multi-casting. Selecting best origin server by connected PoP

### ISSUE THREAD

**Issue by michaelwu0505** [cloudflare/cloudflared#674]
**Describe the feature you'd like**
Cloudflare tunnel equivalent of multi-casting.

When there are multiple origin servers located at geographically far apart  locations, each connected to a near by Cloudflare PoP, I would like the incoming request to be directed to the nearest PoP.

For example, if there are two origin servers. One is at the US, which is connected to LAX. Another is at Singapore and is connected to SIN. When there is a client coming from Japan, I would expect the request is forwarded to the Singapore origin through SIN PoP.

**Describe alternatives you've considered**
Alternative is to not use Cloudflare tunnel and use origin IP that supports multi-casting.

**Comment by abelinkinbio** [comment]
@michaelwu0505 thanks for submitting this feature request. Would you mind sharing a little bit more about what you're aiming to accomplish with this architecture? Also, what does your current deployment look like today? 

Generally, we recommend deploying `cloudflared` as "close" to your origin server as possible to reduce latency. However, in this example, it sounds like you may be leveraging either replicas or multiple tunnels. Have you considered utilizing our [load balancing](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/lb/) solution to intelligently steer traffic towards your origin servers in distinct geographies?

**Comment by michaelwu0505** [comment]
Yes, I have have replication setup of two origin servers: One server connects to KIX/NRT PoP in Japan,  and the other server connects to SIN PoP in Singapore.

Current behavior of Cloudflare tunnel distributes incoming traffic in a round-robin fashion between all connected  tunnels. For example, if there are 8 connections to KIX PoP and 2 connection to SIN PoP, the traffic will be split in 4:1 to KIX and SIN. I believe a helpful feature is for Cloudflare tunnel to send incoming traffic to a PoP that is closer to the client. This feature can be made configurable.

Since I have no other use of a load balancer, I am not considering using one as it will incur additional cost. Thank you for the suggestion.
